### ISSUE TITLE
gnutls 'post-install step did not complete successfully' on 10.12.5

### ISSUE THREAD

**Issue by ralexx** [Homebrew/homebrew-core#14529]
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? 
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
<hr>

> - What you were trying to do (and why)

Performing a routine `brew upgrade`

> - What happened (include command output)
```
==> Upgrading gnutls
==> Downloading https://homebrew.bintray.com/bottles/gnutls-3.5.13.sierra.bottle
######################################################################## 100.0%
==> Pouring gnutls-3.5.13.sierra.bottle.tar.gz
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall gnutls`
==> Summary
🍺  /usr/local/Cellar/gnutls/3.5.13: 1,105 files, 7.6MB
```

> - What you expected to happen

Any post-installation linking etc. should have completed automatically

> - Step-by-step reproduction instructions (by running `brew install` commands)

```
⋊> ~ brew doctor                                                        10:09:42
Your system is ready to brew.
⋊> ~ brew gist-logs gnutls                                              10:26:59
Error: No logs.
⋊> ~ brew update                                                        10:31:54
Already up-to-date.
⋊> ~ brew upgrade gnutls                                                10:39:55
Error: gnutls 3.5.13 already installed
⋊> ~ brew postinstall gnutls                                            10:40:10
==> Postinstalling gnutls
==> Using the sandbox
Error: undefined method `success?' for nil:NilClass
Please report this bug:
  http://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gnutls.rb:74:in `block in post_install'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gnutls.rb:68:in `select'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gnutls.rb:68:in `post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1029:in `block in run_post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:887:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1028:in `run_post_install'
/usr/local/Homebrew/Library/Homebrew/postinstall.rb:16:in `<main>'
```

### config
```
⋊> ~ brew config
HOMEBREW_VERSION: 1.2.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54e01d76db34a3a649c79b189873bfd0a52dbc9f
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fe2cb12124c3657a7b3f4181308576c5dc92e25f
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: 12-core 64-bit arrandale
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.13.1 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_131
macOS: 10.12.5-x86_64
Xcode: 8.3.3
CLT: 8.3.2.0.1.1492020469
X11: 2.7.11 => /opt/X11
```

**Comment by jtokoph** [comment]
same issue when installing openss:
```
brew postinstall openssl
==> Postinstalling openssl
==> Using the sandbox
Error: undefined method `success?' for nil:NilClass
Please report this bug:
  http://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb:94:in `block in post_install'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb:88:in `select'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb:88:in `post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1029:in `block in run_post_install'
/usr/local/Homebrew/Library/Homebrew/formula.rb:887:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1028:in `run_post_install'
/usr/local/Homebrew/Library/Homebrew/postinstall.rb:16:in `<main>'
```
```
HOMEBREW_VERSION: 1.2.2
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 54e01d76db34a3a649c79b189873bfd0a52dbc9f
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fe2cb12124c3657a7b3f4181308576c5dc92e25f
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.13.0 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /Users/jtokoph/.rvm/rubies/ruby-2.3.3/bin/ruby
Java: 1.8.0_121, 1.8.0_31, 1.6.0_65-b14-468
macOS: 10.12.5-x86_64
Xcode: 8.3.3
CLT: 8.3.2.0.1.1492020469
X11: 2.7.9 => /opt/X11
```

**Comment by jtokoph** [comment]
Looks to be related to the rubocop warning updates:
https://github.com/Homebrew/homebrew-core/commits/master?author=MikeMcQuaid

Does 'english' need to be required? https://github.com/bbatsov/rubocop/issues/1747

@MikeMcQuaid

**Comment by bmw** [comment]
Post-install failures from openssl are causing [our macOS Travis tests](https://travis-ci.org/certbot/certbot/jobs/242114627) to fail.

**Comment by MikeMcQuaid** [comment]
Please `brew update`; this is fixed by Homebrew 1.2.3 (just released).
